internal/cloudapi/v2/handler.go

```diff
@@ -501,8 +502,8 @@ func imageTypeFromApiImageType(it ImageTypes, arch distro.Arch) string {
 	return ""
 }
 
-func (h *apiHandlers) GetComposeStatus(ctx echo.Context, id string) error {
-	jobId, err := uuid.Parse(id)
+func (h *apiHandlers) GetComposeStatus(ctx echo.Context, id types.UUID) error {
+	jobId, err := uuid.Parse(string(id))
```

Comment:
Sorry for only reviewing this now. Generated code will return a 400 in case it's an invalid ID here, but also with a bad error message/not an api error. So we either need to add a generic 400 case to the echoError -> apiError conversion in the error handler or force using a string for id.

Reply:
Could do something like https://github.com/Gundersanne/osbuild-composer/commit/6a663c6ac6444719cd1acb2229545386a53cc5bb ; or we set the type explicitly to string so we can return ErrorInvalidComposeId like before, as the translating of these errors do make it less readable :/